### PR TITLE
fix(vscode): Convert the authentication type from Raw to MSI when deploying Logic App

### PR DESCRIPTION
**Pull request by JoaquimMalcampo**
## Requirement Checklist

* [ x] The commit message follows our guidelines
* [x ] Tests for the changes have been added (for bug fixes or features)
* [ ] Docs have been added or updated (for bug fixes or features)

## Type of Change

* [x ] Bug fix
* [ ] Feature
* [ ] Other

## Current Behavior
When deploying Logic App, changes the authentication field in connection to MSI.

## New Behavior
If "Parameterize Connections in Project Load" is enabled or null, then the authentication in parameters.json will convert to MSI. Otherwise, the authentication will convert to MSI in-line in connections.json instead (this will preserve any other parameterization if any).

## Screenshots or Videos (if applicable)

https://github.com/user-attachments/assets/c97db395-4908-4f7b-b786-08f30e26ad48

### CODE REVIEW

**Review by ccastrotrejo — APPROVED** [review]

File: apps/vs-code-designer/src/app/commands/deploy/deploy.ts [hunk 1]
```diff
@@ -332,41 +365,14 @@ async function getProjectPathToDeploy(
...
   }
   return undefined;
 }
-
-async function cleanAndRemoveDeployFolder(deployProjectPath: string): Promise<void> {
-  await fse.emptyDir(deployProjectPath);
-  fse.rmdirSync(deployProjectPath);
-}
-
-async function checkAADDetailsExistsInAppSettings(node: SlotTreeItem, identityWizardContext: IIdentityWizardContext): Promise<boolean> {
```
Comment (mireedmsft) [inline apps/vs-code-designer/src/app/commands/deploy/deploy.ts:355]: nit: Is there a reason why this moved in the file? would be less churn if it stayed in place?

Comment (mireedmsft) [reply apps/vs-code-designer/src/app/commands/deploy/deploy.ts:355]: it looks like there aren't any changes to the method itself

File: apps/vs-code-designer/src/app/commands/deploy/deploy.ts [hunk 2]
```diff
@@ -288,19 +313,27 @@ async function getProjectPathToDeploy(
...
   let parametizedConnections: ConnectionsData;
-
   const targetAppSettings = await node.getApplicationSettings(identityWizardContext as IDeployContext);
-  const resolutionService = new ResolutionService(parametersJson, targetAppSettings);
+
   try {
     parametizedConnections = JSON.parse(connectionsJson);
+
+    if (parametizedConnections.managedApiConnections && Object.keys(parametizedConnections.managedApiConnections).length) {
+      for (const referenceKey of Object.keys(parametizedConnections.managedApiConnections)) {
```
Comment (mireedmsft) [inline apps/vs-code-designer/src/app/commands/deploy/deploy.ts:322]: are there any unit tests for this?

File: apps/vs-code-designer/src/app/commands/deploy/deploy.ts [hunk 3]
```diff
@@ -332,41 +365,14 @@ async function getProjectPathToDeploy(
...
-          type: 'ManagedServiceIdentity',
-        };
       }
+
       settingsToExclude.push(`${referenceKey}-connectionKey`);
     }
-
-    await writeFormattedJson(connectionsFilePath, parametizedConnections);
+    await writeFormattedJson(connectionsFilePathDeploy, parametizedConnections);
+    await writeFormattedJson(parametersFilePathDeploy, parametersJson);
```
Comment (mireedmsft) [inline apps/vs-code-designer/src/app/commands/deploy/deploy.ts:373]: Is this what Jaden's code was missing?

File: apps/vs-code-designer/src/app/commands/deploy/deploy.ts (hunk 1 above)
Comment (ccastrotrejo) [reply apps/vs-code-designer/src/app/commands/deploy/deploy.ts:355]: @JoaquimMalcampo Thats right, if there aren't any changes on this, please avoid just moving code.

Comment (JoaquimMalcampo) [reply apps/vs-code-designer/src/app/commands/deploy/deploy.ts]: reverted

File: apps/vs-code-designer/src/app/commands/deploy/deploy.ts (hunk 2 above)
Comment (JoaquimMalcampo) [reply apps/vs-code-designer/src/app/commands/deploy/deploy.ts]: I did not see any unit tets for the other commands so I did not include any.

**Pull request merged**